### ISSUE TITLE
[Bug] When type = line and the x-axis has more than one of the same data, the linking order does not follow the order of the incoming data.

### ISSUE THREAD

**Issue by rick-chou** [apache/echarts#19062]
### Version

5.4.3

### Link to Minimal Reproduction

https://echarts.apache.org/examples/en/editor.html?c=line-simple&code=PYBwLglsB2AEC8sDeAoWszGAG0iAXMmuhgE4QDmFApqYQOQCGAHhAM73EC-ANMReQAmhVCVjZqAMzAMAzAFJ6fMeQoALGbHoBWRcpIAjYGEwBbBgEZdS4ugDGMMIwjQAMowPVshMKQCu1Nz6mDhGzCK2sJLUjGB-pNQRYuiCsYwAWljmRMkkAJ4AgqxsAJLQgtThWtAw1Jy5vJHoCWyYCSKNuWyMAG7UBaWmjDQdkZ0kqhpyikHEqU6ZwNkA2pGiyWB5IIlabNgQFaQ2uZIQuLQAssAVDADuMQDWAGJnYLTHyWxqwLcAImkAZTUjEEPx8_kCYi4xAAuvoJDRykkSF8wVFGNg2JDYONmEV2Mj0JttgxIKY6vp0Cx2O5PN4cp9vrcLi5aV5wQFKSimRcWGz6b4AmNZuhCsVCRgtjt6D0MQEPuhoIxyQxTAYQBwRbAseRqGxCKsxOsxMTpftoBSmtrTFgwGoOdQuYrldKegdqMBYHqybFqIJYAZGOVbgc7QqSPNGAarehlhYAGwAdgADAAmWSyeOyZMAFgsyZ4sBzADpk7JU3CY7A40m0xms8ntPGLIWS2WK06SDWU-nM9ntImW0XS-XK7lYwme_Xs_HtKnWyOO1Xu3W-wnkwXh-2x-OV734zns-WF9vOxPa_vD8n4_PYBZS8mABw73J7-tXm-F2Sl1M5l_JN9Mw_W821HM9q0nOsc0faDH1_E9RytGFhXQZCuAAbiAA

### Steps to Reproduce

```js
option = {
  tooltip: {
    trigger: 'axis'
  },
  grid: {
    left: '3%',
    right: '5%',
    bottom: '15%',
    containLabel: true
  },
  toolbox: {
    feature: {
      dataZoom: {
        yAxisIndex: 'none'
      },
      restore: {},
      saveAsImage: {}
    },
    right: '3%'
  },
  dataZoom: [
    {
      type: 'slider',
      filterMode: 'weakFilter',
      showDataShadow: true
    }
  ],
  legend: {
    show: false
  },
  xAxis: {
    type: 'time',
    axisLabel: {
      showMinLabel: true,
      showMaxLabel: true
    }
  },
  yAxis: {
    type: 'value',
    name: 'mbps'
  },
  series: [
    {
      type: 'line',
      smooth: true,
      name: 'video estimated bandwidth',
      data: [
        [1670233630410, 4.032],
        [1670233630561, 4.032],
        [1670233630571, 4.032],
        [1670233630652, 4.032],
        [1670233631600, 4.032],
        [1670233643032, 4.032],
        [1670233643062, 1.008],
        [1670233643062, 3.024],
        [1670233643062, 4.032],
        [1670234848824, 4.032]
      ]
    }
  ]
};
```

### Current Behavior

When type = line and the x-axis has more than one of the same data, the linking order does not follow the order of the incoming data.

### Expected Behavior

In my case, 

```
  series: [
    {
      type: 'line',
      smooth: true,
      name: 'video estimated bandwidth',
      data: [
        [1670233630410, 4.032],
        [1670233630561, 4.032],
        [1670233630571, 4.032],
        [1670233630652, 4.032],
        [1670233631600, 4.032],
        [1670233643032, 4.032],-------------------------------------
        [1670233643062, 1.008], ---↑ they should be connected.      ↑
        [1670233643062, 3.024],                                     ↑
        [1670233643062, 4.032], ------------------------------------↑ but they are connected
        [1670234848824, 4.032] 
      ]
    }
  ]
```

### Environment

```markdown
- OS: Macos 13.5 (22G74)
- Browser: Chrome 116.0.5845.140 (Official Build) (arm64)
- Framework: React@18
```

**Comment by parteekmalik** [comment]
does you want it to be like this:
![Screenshot 2023-09-05 111534](https://github.com/apache/echarts/assets/64765095/e016af28-e012-4e16-ab60-fa3eb1573184)
if yes then it was happening becase of data[][0] has multiple same value
i changed data to :
data: [
        [1, 4.032],
        [2, 4.032],
        [3, 4.032],
        [4, 4.032],
        [5, 4.032],
        [6, 4.032],
        [7, 1.008],
        [8, 3.024],
        [9, 4.032],
        [10, 4.032]
      ]

**Comment by rick-chou** [comment]
Yeah, I know in this case, the chart is correct. But in my needs, there will be the same data, under this premise, the data connection order of the chart is not correct, but the data order on the tooltip is correct

**Comment by rick-chou** [comment]
the order of the data on tooltip is correct
<img width="812" alt="Screenshot 2023-09-05 at 21 18 11" src="https://github.com/apache/echarts/assets/56244178/81001cda-c339-4aec-bc35-1060c469e70d">

**Comment by parteekmalik** [comment]
you can use a function to add value of same key 

function sumValuesByKey(data: number[][]): number[][] {
  const resultMap = new Map<number, number>();

  for (const [key, value] of data) {
    if (resultMap.has(key)) resultMap.set(key, resultMap.get(key)! + value);
    else resultMap.set(key, value);
  }

  const resultArray: number[][] = [];
  for (const [key, value] of resultMap) {
    resultArray.push([key, value]);
  }

  return resultArray;
}
